### PR TITLE
Fixes panic when id not found in collector

### PR DESCRIPTION
**Pull request by seans3**
* Fixes panic in `ResourceStateCollector.processStatusEvent` caused by de-referencing a field from a nil pointer.
* Fixes: https://github.com/GoogleContainerTools/kpt/issues/1329
* Adds unit tests for `collector.go` increasing unit test coverage from 16.1% to 23.7%
* Adds logging messages for debugging

### CODE REVIEW

File: cmd/printers/table/collector.go [hunk 1]
```diff
@@ -182,8 +183,17 @@ func (r *ResourceStateCollector) processEvent(ev event.Event) error {
 // update for a resource.
 func (r *ResourceStateCollector) processStatusEvent(e event.StatusEvent) {
 	if e.Type == event.StatusEventResourceUpdate {
+		klog.V(7).Infoln("processing status event")
 		resource := e.Resource
-		previous := r.resourceInfos[resource.Identifier]
+		if resource == nil {
```
Comment (mortent) [inline cmd/printers/table/collector.go:188]: So making sure it doesn't crash is good, but do we know why this happened in the first place? The idea is that the `StatusEventResourceUpdate` is about a specific resource, so why doesn't the event include the resource?

File: cmd/printers/table/collector.go [hunk 2]
```diff
@@ -182,8 +183,17 @@ func (r *ResourceStateCollector) processEvent(ev event.Event) error {
...
+			klog.V(4).Infoln("status event missing Resource field; no processing")
+			return
+		}
+		previous, found := r.resourceInfos[resource.Identifier]
+		if !found {
```
Comment (mortent) [inline cmd/printers/table/collector.go:193]: Similar to above, but this seems like a situation that is a bug. If I remember correctly the `resourceInfos` map should be pre-populated with all resources which we are interested in. So if we get an event about a different resource, something must be wrong elsewhere.

File: cmd/printers/table/collector.go (hunk 1 above)
Comment (seans3) [reply cmd/printers/table/collector.go:188]: I don't think this is happening in this bug. But we should program defensively.

File: cmd/printers/table/collector.go (hunk 2 above)
Comment (seans3) [reply cmd/printers/table/collector.go:193]: This probably happens when a `PersistentVolume` status hangs, but I haven't been able to reproduce this yet. Please have a look at the connected issue, which points out this `previous` is nil and not found.

**Pull request merged**